Zilon.Core/Zilon.Core/World/IGlobeInitializer.cs

```diff
@@ -0,0 +1,472 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Zilon.Core.MapGenerators;
+using Zilon.Core.PersonGeneration;
+using Zilon.Core.Persons;
+using Zilon.Core.Players;
+using Zilon.Core.Schemes;
+using Zilon.Core.Tactics;
+using Zilon.Core.Tactics.Behaviour;
+
+namespace Zilon.Core.World
+{
+    public interface IGlobeInitializer
+    {
+        Task<IGlobe> CreateGlobeAsync(string startLocationSchemeSid);
+    }
+
+    public interface IGlobe
+    {
+        void AddSectorNode(ISectorNode sectorNode);
+
+        IEnumerable<ISectorNode> SectorNodes { get; }
+
+        Task UpdateAsync();
+    }
+
+    public interface IGlobeTransitionHandler
+    {
+        Task ProcessAsync(IGlobe globe, ISector sector, IActor actor, RoomTransition transition);
+    }
+
+    public sealed class GlobeTransitionHandler : IGlobeTransitionHandler, IDisposable
+    {
+        private readonly IGlobeExpander _globeExpander;
+
+        private readonly SemaphoreSlim _semaphoreSlim;
+
+        public GlobeTransitionHandler(IGlobeExpander globeExpander)
+        {
+            _globeExpander = globeExpander ?? throw new ArgumentNullException(nameof(globeExpander));
+
+            //Instantiate a Singleton of the Semaphore with a value of 1. This means that only 1 thread can be granted access at a time.
+            _semaphoreSlim = new SemaphoreSlim(1, 1);
+        }
+
+        public async Task ProcessAsync(IGlobe globe, ISector sector, IActor actor, RoomTransition transition)
+        {
+            if (globe is null)
+            {
+                throw new ArgumentNullException(nameof(globe));
+            }
+
+            if (sector is null)
+            {
+                throw new ArgumentNullException(nameof(sector));
+            }
+
+            if (actor is null)
+            {
+                throw new ArgumentNullException(nameof(actor));
+            }
+
+            if (transition is null)
+            {
+                throw new ArgumentNullException(nameof(transition));
+            }
+
+            var sectorNode = transition.SectorNode;
+
+            //TODO Разобраться с этим кодом.
+            // https://blog.cdemi.io/async-waiting-inside-c-sharp-locks/
+            //Asynchronously wait to enter the Semaphore. If no-one has been granted access to the Semaphore, code execution will proceed, otherwise this thread waits here until the semaphore is released 
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task
+            await _semaphoreSlim.WaitAsync();
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=6105716)